core/runtime.rs

```diff
@@ -349,6 +357,45 @@ impl JsRuntime {
       .unwrap();
   }
 
+  /// Initializes JS of provided Extensions
+  // NOTE: this will probably change when streamlining snapshot flow
+  pub fn init_extension_js(&mut self) -> Result<(), AnyError> {
+    // Take extensions to avoid double-borrow
+    let mut extensions: Vec<Extension> = self.extensions.drain(..).collect();
```

Comment:
This will needlessly allocate & deallocate a second Vec for the extensions, you can use `std::mem::replace` to put an empty Vec (doesn't allocate) into its place instead and then replace it back once the call is done.

```rust
let mut extensions = std::mem::replace(&mut self.extensions, Vec::new()); // Allocation free
for m in extensions.iter_mut() {
    // ...
}
std::mem::replace(&mut self.extensions, extensions); // Put the extensions back.
```

Reply:
Sure, that works too. Keep in mind that this code is very much not in a hot loop, so clarity matters more here than a small extra alloc (I'm fine with `std::mem::*` but it seems that it can mislead people to thinking that we're doing something low level and they need to pay careful attention)